plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/BannedElement.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.validation;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * A blacklisted element found in appengine-web.xml
+ */
+public class BannedElement {
+
+  private final String message;
+  private final DocumentLocation start;
+  private final int length;
+  
+  public BannedElement(String elementName, DocumentLocation start, int length) {
+    Preconditions.checkNotNull(elementName, "elementName is null");
+    Preconditions.checkNotNull(start, "start is null");
+    this.message = AppEngineWebBlacklist.getBlacklistElementMessage(elementName);
```

Comment:
What if the elementName is not a banned element's name (i.e. `message` is `""`). Shouldn't we guard against it?